### PR TITLE
Create canary job to run new e2e image

### PR DESCRIPTION
**Pull request by fejta**

### CODE REVIEW

File: jobs/ci-kubernetes-e2e-gce-canary.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Everything else is the same as the other job it is testing
+BOOTSTRAP_METADATA_CANARY_IMAGE=true
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-canary.env]: I cannot find this `BOOTSTRAP_METADATA_CANARY_IMAGE` anywhere else? It's just some env you want to define?

File: jobs/ci-kubernetes-e2e-gce-canary.env [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+KUBEKINS_TIMEOUT=50m
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-e2e-gce-canary.env]: this is the same as `ci-kubernetes-e2e-gce.env`, get rid of it?

File: jobs/ci-kubernetes-e2e-gce-canary.env (hunk 1 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-canary.env]: This is just metadata. `BOOTSTRAP_METADATA_FOO=bar` will result in adding a `{"foo": "bar"}` entry into the metadata.json dictionary.

File: jobs/ci-kubernetes-e2e-gce-canary.env (hunk 2 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-canary.env]: Done. It is running fewer tests so I want to redefine this here (and it also needs to exist to pass unit tests).

File: jobs/ci-kubernetes-e2e-gce-canary.env (hunk 1 above)
Comment (fejta) [inline jobs/ci-kubernetes-e2e-gce-canary.env]: I'm going to just drop this var, might add it back in the scenario.

**Pull request merged**